### ISSUE TITLE
Don't be tied hard to new Tensorflow/PyTorch versions

### ISSUE THREAD

**Issue by bfirsh** [replicate/cog#181]
I just have `tensorflow==2.5.0` in my `cog.yaml` and this shouldn't fail hard:

```
ⅹ Failed to generate Dockerfile: No matching tensorflow CPU package for version 2.5.0
```

It doesn't need to do anything for CPU packages (I think?) and even if it did, it should be a warning not a hard failure.

**Comment by bfirsh** [comment]
This has been fixed for PyTorch in #229, but still an issue for Tensorflow I believe.

**Comment by bfirsh** [comment]
It might be worth stepping back and think how we design this from a user's point of view. There are two cases where a user might be selecting a PyTorch version which is different to what Cog allows you do to:

1. They know what they are doing, and selecting a version that Cog doesn't support. Either Cog is doing something wrong, or they want to use a PyTorch version which isn't supported by Cog, or they're doing something else custom.
2. They are doing something wrong.

In the (1) case, they are doing something very intentional. In (2), they are not. The problem with displaying warnings, like we have done in #229 and #269 is that user (1) is fine, but user (2) might not see the warnings and then something is broken for some reason.

Perhaps Cog should fail hard when it fails to resolve versions, and then have some way that a user explicitly says "I know what I'm doing" and then it doesn't fail hard. Some `cog.yaml` option to disable magic version resolving, or something.

Either way, this deserves a bit more thought.
